include/boost/astronomy/coordinate/arithmetic.hpp

```diff
@@ -120,6 +100,85 @@ auto cross
         >(result);
 }
 
+//!Returns the cross product with Spherical/Spherical-Equatorial representation as first argument
+template
+<
+    template<typename...> class Representation,
+    typename Other,
+    typename ...Args
+>
+auto cross
+(
+    Representation<Args...> const& vector1,
+    Other const& vector2
+)
+{
+    auto representation1 = make_cartesian_representation(vector1);
+    auto representation2 = make_cartesian_representation(vector2);
+
+    typedef decltype(representation1) representation1_type;
+    typedef decltype(representation2) representation2_type;
+
+    bg::model::point
+    <
+        typename std::conditional
+        <
+            sizeof(typename representation2_type::type) >=
+                sizeof(typename representation1_type::type),
+            typename representation2_type::type,
+            typename representation1_type::type
+        >::type,
+        3,
+        bg::cs::cartesian
+    > tempPoint1, tempPoint2, result;
+
+    bg::transform(representation1.get_point(), tempPoint1);
+    bg::transform(representation2.get_point(), tempPoint2);
+
+    bg::set<0>(result, (bg::get<1>(tempPoint1)*bg::get<2>(tempPoint2)) -
+        ((bg::get<2>(tempPoint1)*
+        bu::conversion_factor(typename representation1_type::quantity3::unit_type(),
+        typename representation1_type::quantity2::unit_type()))*
+        (bg::get<1>(tempPoint2)*
+        bu::conversion_factor(typename representation2_type::quantity2::unit_type(),
+        typename representation2_type::quantity3::unit_type()))));
+
+    bg::set<1>(result, (bg::get<2>(tempPoint1)*bg::get<0>(tempPoint2)) -
+        ((bg::get<0>(tempPoint1)*
+        bu::conversion_factor(typename representation1_type::quantity1::unit_type(),
+        typename representation1_type::quantity3::unit_type()))*
+        (bg::get<2>(tempPoint2)*
+        bu::conversion_factor(typename representation2_type::quantity3::unit_type(),
+        typename representation2_type::quantity1::unit_type()))));
+
+    bg::set<2>(result, (bg::get<0>(tempPoint1)*bg::get<1>(tempPoint2)) -
+        ((bg::get<1>(tempPoint1)*
+        bu::conversion_factor(typename representation1_type::quantity2::unit_type(),
+        typename representation1_type::quantity1::unit_type()))*
+        (bg::get<0>(tempPoint2)*
+        bu::conversion_factor(typename representation2_type::quantity1::unit_type(),
+        typename representation2_type::quantity2::unit_type()))));
+
```

Comment:
I think code snippet from line 138 to line 160 should be changed. Here conversion are not matching with the quantity returning in the return statement.